druid/src/widget/image.rs

```diff
@@ -438,4 +446,104 @@ mod tests {
             },
         );
     }
+
+    #[test]
+    fn width_bound_paint() {
+        use crate::{tests::harness::Harness, widget::Scroll, WidgetId};
+        let _id_1 = WidgetId::next();
+        let image_data = ImageData {
+            pixels: vec![255, 255, 255, 0, 0, 0, 0, 0, 0, 255, 255, 255],
+            x_pixels: 2,
+            y_pixels: 2,
+            format: ImageFormat::Rgb,
+        };
+
+        let image_widget =
+            Image::new(image_data).interpolation_mode(InterpolationMode::NearestNeighbor);
+
+        Harness::create_with_render(
+            true,
+            Scroll::new(image_widget).vertical(),
+            Size::new(400., 400.),
+            |harness| {
+                harness.send_initial_events();
+                harness.just_layout();
+                harness.paint();
+            },
+            |target| {
+                // the width should be calculated to be 400.
+                let width = 400;
+                let raw_pixels = target.into_raw();
+                assert_eq!(raw_pixels.len(), 400 * width * 4);
+
+                // Being a height bound widget every row will have no padding at the start and end._id_1
+
+                // The image starts at (0,0), so 200 white and then 200 black.
```

Comment:
I find the comment a bit misleading, because you're actually testing only the last white-black row.